### PR TITLE
Add axis order for plot_pareto_front

### PR DESCRIPTION
**Pull request by okdshin**
## Motivation

Current `plot_pareto_front` function can not transpose axis.

## Description of the changes

Add `axis_order` argument by which users can handle axis order. It takes a list of index of axes.

### CODE REVIEW

**Review by crcrpar — COMMENTED** [review]
Thank you for the patch you made!

I left some comments that require some cumbersome work, but I think they'd make this patch better. 🙂

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 1]
```diff
@@ -142,18 +163,29 @@ def _get_pareto_front_3d(
         point_colors += ["red"] * len(non_pareto_trials)
         trials += non_pareto_trials
 
+    if axis_order is None:
+        axis_order = list(range(3))
+    elif max(axis_order) > 2:
+        _logger.warning("axis_order contains invalid index higher than 2")
+    elif min(axis_order) < 0:
+        _logger.warning("axis_order contains invalid index lower than 0")
```
[the remark this answers is not in the document]
Comment (okdshin) [reply optuna/multi_objective/visualization/_pareto_front.py:176]: OK. I will modify.

**Review by crcrpar — COMMENTED** [review]
What a quick action. Much appreciated!

Could you add tests for the newly added exceptions though they may be optional?

**Review by crcrpar — APPROVED** [review]
thank you. Looks good to me.

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 2]
```diff
@@ -112,20 +121,47 @@ def _get_pareto_front_2d(
         point_colors += ["red"] * len(non_pareto_trials)
         trials += non_pareto_trials
 
+    if axis_order is None:
+        axis_order = list(range(2))
+    else:
+        if len(axis_order) != 2:
+            raise ValueError(
+                f"Size of `axis_order` {axis_order}. Expect: 2, Actual: {len(axis_order)}"
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:129]: Again a detail but we usually place a period.
```suggestion
                f"Size of `axis_order` {axis_order}. Expect: 2, Actual: {len(axis_order)}."
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 3]
```diff
@@ -112,20 +121,47 @@ def _get_pareto_front_2d(
...
+            )
+        if np.unique(axis_order).size != 2:
+            raise ValueError(f"Elements of given `axis_order` {axis_order} are not unique!")
+        if max(axis_order) > 1:
+            raise ValueError(
+                f"Given `axis_order` {axis_order} contains invalid index {max(axis_order)} "
+                "higher than 1"
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:136]: Ditto.
```suggestion
                "higher than 1."
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 4]
```diff
@@ -112,20 +121,47 @@ def _get_pareto_front_2d(
...
+            )
+        if min(axis_order) < 0:
+            raise ValueError(
+                f"Given `axis_order` {axis_order} contains invalid index {min(axis_order)} "
+                "lower than 0"
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:141]: Ditto.
```suggestion
                "lower than 0."
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 5]
```diff
@@ -62,6 +65,9 @@ def objective(trial):
             "Objective {objective_index}" is used instead.
         include_dominated_trials:
             A flag to include all dominated trial's objective values.
+        axis_order:
+            A list of axis index indicating the order of axis. If :obj:`None` is specified,
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:69]: Another minor.
```suggestion
            A list of indices indicating the axis order. If :obj:`None` is specified,
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 6]
```diff
@@ -142,18 +178,42 @@ def _get_pareto_front_3d(
         point_colors += ["red"] * len(non_pareto_trials)
         trials += non_pareto_trials
 
+    if axis_order is None:
+        axis_order = list(range(3))
+    else:
+        if len(axis_order) != 3:
+            raise ValueError(
+                f"Size of `axis_order` {axis_order}. Expect: 3, Actual: {len(axis_order)}"
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:186]: Ditto.
```suggestion
                f"Size of `axis_order` {axis_order}. Expect: 3, Actual: {len(axis_order)}."
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 7]
```diff
@@ -142,18 +178,42 @@ def _get_pareto_front_3d(
...
+            )
+        if np.unique(axis_order).size != 3:
+            raise ValueError(f"Elements of given `axis_order` {axis_order} are not unique!")
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:189]: Ditto.
```suggestion
            raise ValueError(f"Elements of given `axis_order` {axis_order} are not unique!.")
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 8]
```diff
@@ -142,18 +178,42 @@ def _get_pareto_front_3d(
...
+        if max(axis_order) > 2:
+            raise ValueError(
+                f"Given `axis_order` {axis_order} contains invalid index {max(axis_order)} "
+                "higher than 2"
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:193]: Ditto.
```suggestion
                "higher than 2."
```

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 9]
```diff
@@ -142,18 +178,42 @@ def _get_pareto_front_3d(
...
+            )
+        if min(axis_order) < 0:
+            raise ValueError(
+                f"Given `axis_order` {axis_order} contains invalid index {min(axis_order)} "
+                "lower than 0"
```
Comment (hvy) [inline optuna/multi_objective/visualization/_pareto_front.py:198]: Ditto.
```suggestion
                "lower than 0."
```

**Review by hvy — CHANGES_REQUESTED** [review]
The design and the logic looks good to me as well. I left some nits.

File: optuna/multi_objective/visualization/_pareto_front.py [hunk 10]
```diff
@@ -2,6 +2,8 @@
 from typing import List
 from typing import Optional
 
+import numpy as np
```
[the remark this answers is not in the document]
Comment (okdshin) [reply optuna/multi_objective/visualization/_pareto_front.py]: OK

**Review by hvy — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**